### PR TITLE
[MMA-14178] Back-Porting Jenkins Semaphore migration changes

### PR DESCRIPTION
**Pull request by VanamalaNarasimhaBhargav**
1. During Jenkins to Semaphore migration service-bot added some changes to an exiting PR confluentinc/control-center-images/pull/67. 
2. Earlier this PR was merged as is but we later realised all changes from this PR were not required so this PR is reverted at confluentinc/control-center-images/pull/75 & only required changes are cherry-picked in this PR.
3. Meanwhile since the changes are reverted service bot raised a different PR with same changes at confluentinc/control-center-images/pull/77 so cherry-picked commits from this instead.
4. **How to verify?** Ensure all changes from .semaphore folder in the PR confluentinc/control-center-images/pull/77 are picked up

Will make sure that the changes are reviewed by c3 & cp-build team.

### CODE REVIEW

**Review by wadhwa1 — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1 @@
+* @confluentinc/c3-team
```
Comment (nagarwal01) [inline .github/CODEOWNERS:1]: it says there is some error, check other codeowners file

Comment (VanamalaNarasimhaBhargav) [reply .github/CODEOWNERS:1]: Same line worked at https://github.com/confluentinc/blueway/blob/master/.github/CODEOWNERS
I am following up with devprod team at https://confluent.slack.com/archives/C09EP1SS3/p1715067008892029
will raise a PR/work seperately to fix this.
FYI this was present in the original PR by service bot as well https://github.com/confluentinc/control-center-images/pull/77/files

Comment (VanamalaNarasimhaBhargav) [reply .github/CODEOWNERS:1]: **UPDATE:** Raised a PR to fix this at https://github.com/confluentinc/admin/pull/853

Comment (VanamalaNarasimhaBhargav) [reply .github/CODEOWNERS:1]: **UPDATE:** This issue is resolved now.

**Pull request merged**